### PR TITLE
ENH/TST: Add BaseMethodsTests tests for ArrowExtensionArray

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm. cc @jbrockmendel for a quick glance

**Comment by jbrockmendel** [comment]
Will take a look today (and start whittling down the ping backlog)

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -988,6 +1014,287 @@ def test_EA_types(self, engine, data, request):
...
+                    reason=(
+                        f"Checking ndim when using arraymanager with "
+                        f"{pa_dtype} and periods={periods}"
+                    )
+                )
+            )
+        super().test_container_shift(data, frame, periods, indices)
+
+    @pytest.mark.xfail(
+        reason="result dtype pyarrow[bool] better than expected dtype object"
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:1254]: medium-term should just override the test with the correct/better behavior

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: pandas/tests/extension/base/methods.py [hunk 2]
```diff
@@ -49,8 +50,7 @@ def test_value_counts_with_normalize(self, data):
         else:
             expected = pd.Series(0.0, index=result.index)
             expected[result > 0] = 1 / len(values)
-
-        if isinstance(data.dtype, pd.core.dtypes.dtypes.BaseMaskedDtype):
+        if na_value_for_dtype(data.dtype) is pd.NA:
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/tests/extension/base/methods.py:53]: Probably similar to https://github.com/pandas-dev/pandas/issues/27825

**Pull request merged**